operations/firmware/Firmware_Upgrade_using_SPP_on_HPE_ProLiant_Servers.md

```diff
@@ -0,0 +1,89 @@
+# Firmware Upgrade using SPP on HPE ProLiant Servers
+
+## Link to download SPP Image
+
+`https://techlibrary.hpe.com/us/en/enterprise/servers/products/service_pack/spp/index.aspx`
+
+* Naming convention example: `2023.04.00.00` is the ISO image released in 4th Month of 2023
+* Download the latest version or the version you require.
+* Firmware can be deployed either individually by component or the SPP ISO image as a whole.
+
+## Deploying Individual Firmware Components
+
+### Get the individual firmware component from SPP ISO image
+
+* Mount or extract the SPP ISO image and locate the `contents.html` file.
+* Get the location of the file/package in the extracted/mounted ISO image from the 'contents.html' file.  Usually it will be in the folder `Packages`.
+* For example, to flash the BIOS/ROM for `DL360 Gen11` server, find DL360 in `Description` column, and then find relevant component to flash. Note the extension should be `.fwpkg` if we are flashing through ILO.
+
+### Steps to deploy individual components
+
+* Login to the ILO page in web browser
+* Go to the `Firmware & OS` Software at the left section
+* Select `upload to iLO Repository`
+* Select the file where it is located and click on upload.
+* Go to iLO Repository, you should be able to see the uploaded component. Click on `Install Component` and then Click on `TPM Override` then click on `Add to the Installation queue`
+* You can check for the status message in Installation Queue. If you see any failed instance in Queue, then delete that instance else installation of next component would not get start if expiration is set to never for the failed instance.
+
+## Deploy SPP ISO image as a Whole
+
+### Pre-requisites
+
+* Downloaded SPP ISO image
+* ILO accessible on system where you are flashing firmware using SPP ISO image
+* Web Browser
+
+The SPP ISO Image can be deployed locally or remotely.
+
+## Deploying the ISO Image using the local method
+
+* Login to ILO in web browser
+* Select `Remote console & Media` on to the left side
+* Launch the remote console either `HTML5` or `.NET` console or through `HPLOCONS`. Best option would be .NET and HPLOCONS
+* Mount the ISO image either local location of the image or URL path. If URL path is given make sure that is in same network as ILO network
+* Do cold reboot or Reset the node from iLO remote console
+* Press `F9` and go to `One time Boot` menu and select the mounted image, if URL is given then you see the URL path.
+* Select `Interactive` option
+* Accept and proceed
+* Click on `Firmware Update` and then Click OK
+* Inventory takes some time, wait until it completes and then click Next
+* Select the components that needs firmware flashing, you can also check for the version by clicking view details for each component.
+If its already up to date then by default it will not be selected, you can force update the component. Click on TPM override and then deploy
+* After deployment, check for the status and reboot the node

```

Comment:
```suggestion
1. Log in to ILO in web browser.
1. Select `Remote console & Media` on to the left side.
1. Launch the remote console either `HTML5` or `.NET` console or through `HPLOCONS`. Best option would be .NET and HPLOCONS.
1. Mount the ISO image either local location of the image or URL path. If URL path is given make sure that is in same network as ILO network.
1. Do a cold reboot or reset the node from the iLO remote console.
1. Press `F9` and go to `One time Boot` menu and select the mounted image. If the URL is given, then you will see the URL path.
1. Select the `Interactive` option.
1. Accept and proceed.
1. Click on `Firmware Update` and then Click `OK`.
1. Inventory takes some time, wait until it completes and then click `Next`.
1. Select the components that needs firmware flashing, you can also check for the version by clicking view details for each component.
If it is already up to date then by default it will not be selected. You can force update the component. Click on `TPM override`, and then deploy.
1. After deployment, check for the status and reboot the node.

```